src/descriptions.cpp

```diff
@@ -0,0 +1,94 @@
+#include "game.h"
+#include "player.h"
+#include "monster.h"
+#include "map.h"
+#include "mapdata.h"
+#include "input.h"
+#include "ui.h"
+
+enum class description_target : int {
+    creature,
+    furniture,
+    terrain
+};
+
+void game::extended_description( const tripoint &p )
+{
+    const int left = 0;
+    const int right = TERMX;
+    const int top = 3;
+    const int bottom = TERMY;
+    const int width = right - left;
+    const int height = bottom - top;
+    WINDOW *w_head = newwin( top, TERMX, 0, 0 );
+    WINDOW *w_main = newwin( height, width, top, left );
+    // @todo De-hardcode
+    std::string header_message = _( "\
+c to describe creatures, f to describe furniture, t to describe terrain, esc/enter to close." );
+    description_target cur_target = description_target::creature;
+    mvwprintz( w_head, 0, 0, c_white, header_message.c_str() );
+
+    // Set up line drawings
+    for( int i = 0; i < TERMX; i++ ) {
+        mvwputch( w_head, top - 1, i, c_white, LINE_OXOX );
+    }
+
+    wrefresh( w_head );
+    int ch = 'c';
+    do {
+        std::string desc;
+        // Allow looking at invisible tiles - player may want to examine hallucinations etc.
+        switch( cur_target ) {
+            case description_target::creature: {
+                const Creature *critter = critter_at( p );
+                if( critter != nullptr && u.sees( *critter ) ) {
+                    desc = critter->extended_description();
+                } else {
+                    desc = _( "You do not see any creature here." );
+                }
+            }
+            break;
```

Comment:
Oh then that's fine. I only found those code style related minor issues, no other obvious problem exists in the code as far as I read. BTW I don't know why my comment got posted twice.